pkgs/development/python-modules/brainflow/default.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, brainflow
+, cmake
+, nptyping
+, numpy
+, typish
+, pythonOlder
+}:
+let
+  pname = "brainflow";
+  version = "unstable-2024-05-01"; # Latest release has broken build
+  source = fetchFromGitHub {
+    owner = "brainflow-dev";
+    repo = "brainflow";
+    #rev = version;
+    rev = "41c304e7545a7465b06201dd4652ef93822edb83";
+    hash = "sha256-wazbL4LGn9aLS5Ek8EW/rt4S+VHyusqTjvZWNBTKS3w=";
+  };
+in
+buildPythonPackage rec {
+  inherit pname version;
+
+  disabled = pythonOlder "3.7";
+
+  src = source + "/python_package";
+
+  propagatedBuildInputs = [
+    numpy
+    # TODO: There's got to be a better way to do this, right?
+    (nptyping.overrideAttrs (oldAttrs: rec {
+      version = "1.4.4";
+      src = fetchFromGitHub {
+        owner = "ramonhagenaars";
+        repo = "nptyping";
+        rev = "refs/tags/v${version}";
+        hash = "sha256-c9Qoufn9m3H03Pc8XhGzTBeixnl/elkalv50OrW4gJY=";
+      };
+
+      propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [
+        typish
+      ];
+
+      disabledTestPaths = [
+        "tests/test_functions/test_get_type.py"
+        "tests/test_functions/test_py_type.py"
+      ];
+    }))
```

Comment:
This is horrible. If there's a way to make this less horrible, let me know, please!

Reply:
currently I don't think there is a better way. 

you can create issue/PR upstream to update code so that it compatible with nptyping  2.x